File: 6 - Hades' Magical Numbers/solution.go
import (
	"fmt"
	"sort"
)

func findMaximumLifeEnergy(powers []int) int {
	maxAns := 1
	for i := 0; i < 5; i++ {
		sort.Ints(powers)
		powers[0]++
	}
	for _, power := range powers {
		maxAns *= power
	}
	return maxAns
}

func main() {
	var t int
	fmt.Scan(&t)
	for t > 0 {
		t--
		powers := make([]int, 3)
		for i := 0; i < 3; i++ {
			fmt.Scan(&powers[i])
		}
		fmt.Println(findMaximumLifeEnergy(powers))
	}
}

File: 6 - Hades' Magical Numbers/solution.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int findMaximumLifeEnergy(int powers[]) {
    int maxAns = 1;
    for (int i = 0; i < 5; i++) {
        sort(powers, powers + 3);
        powers[0]++;
    }
    for (int j = 0; j < 3; j++) {
        maxAns *= powers[j];
    }
    return maxAns;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int powers[3];
        for (int i = 0; i < 3; i++) {
            cin >> powers[i];
        }
        cout << findMaximumLifeEnergy(powers) << endl;
    }
    return 0;
}
